src/components/buttons/button-accessibility-image-alt.jsx

```diff
@@ -0,0 +1,163 @@
+import ButtonIcon from './button-icon.jsx';
+import ButtonProps from '../base/button-props.js';
+import ButtonStateClasses from '../base/button-state-classes.js';
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+/**
+ * The ButtonAccessibilityImageAlt class provides functionality for changing text color in a document.
+ *
+ * @class ButtonAccessibilityImageAlt
+ * @uses ButtonProps
+ * @uses ButtonStateClasses
+ *
+ */
+class ButtonAccessibilityImageAlt extends React.Component {
+    static key = 'AccessibilityImageAlt';
+
+    static propTypes = {
+        /**
+         * The editor instance where the component is being used.
+         *
+         * @instance
+         * @memberof ButtonAccessibilityImageAlt
+         * @property {Object} editor
+         */
+        editor: PropTypes.object.isRequired
+    };
+
+
+    constructor(prop, context) {
+        super(prop, context);
+
+        const image = this.props.editor.get('nativeEditor').getSelection().getSelectedElement();
+
+        this.state = {
+            element: image,
+            imageAlt: image ? image.getAttribute('alt') : ''
+        };
+    }
+
+    /**
+     * Lifecycle. Renders the UI of the button.
+     *
+     * @method render
+     * @return {Object} The content which should be rendered.
+     */
+    render() {
+        var cssClass = 'ae-button ' + this.getStateClasses();
+
+        if (this.props.renderExclusive) {
+            return (
+                <div className="ae-container-edit-link">
+                    <div className="ae-container-input xxl">
+                        <input aria-label="alt" className="ae-input" onChange={this._handleAltChange.bind(this)} onKeyDown={this._handleKeyDown.bind(this)} placeholder="alt" ref="refAltInput" title="alt" type="text" value={this.state.imageAlt}></input>
```

Comment:
In general the places you're doing `this._something.bind(this)` (etc), it would probably be better to make those arrow functions on instance properties. eg. change `onChange={this._handleAltChange.bind(this)}` to `onChange={this._handleAltChange}` and change the method definition from:

```javascript
_handleAltChange() {
  // ...
}
```
to:

```javascript
_handleAltChange = () => {
  // ...
};
```

This avoids rebinding the function on every render and in general is considered a good practice because it means that the callbacks are `===` if they ever compared (for example, as props in `shouldComponentUpdate`). Even if you think the callback will never be used in this way, it is in general a good rule of thumb is to not really see `bind` anywhere in a React component.

Reply:
@wincent yes I was actually thinking the same but didn't do it to avoid many changes, but since we're already re-writing stuff up it's prbably a good idea to go with that now.

Thanks!